### ISSUE TITLE
[question] Running on low resource devices

### ISSUE THREAD

**Issue by kirk86** [allo-/virtual_webcam_background#83]
Hi, I don't have time to do a proper pull request, but I've written a detailed [blog post](https://kirk86.github.io/2022/02/virtual-webcam-background/) to run this project on intel-based macbook pro 2015.

I was wondering if you could incorporate part of it in the main repo so that it works out of the box in OSX, for instance you could detect if the code runs on OSX and execute the equivalent code, or maybe add it in the wiki? If that makes sense?

**Comment by allo-** [comment]
Thank you for the nice howto!

I'll need to diff the files and I wonder why you shortened the requirements.txt and if this will work on other machines.
It looks like some code changes can be included. The run script should probably stay as it is and the preparation steps be included in the documentation as for example the virtual cam only needs to be added once per boot.

And I would like to ask you, if I can add your article as markdown to the project, so the documentation is bundled with the code.

**Comment by kirk86** [comment]
> why you shortened the requirements.txt

Since in the howto the 2nd step installs `scipy` which brings automatically `numpy` and other packages as prerequisites when using conda. I think the same thing will happen if we only install `tensorflow` it will bring `numpy` and other stuff as prereq. in conda.

> if this will work on other machines

I presume yes, to be on the safe side, just use `Actions` on this repo to create VMs for instance for OSX, Linux, Windows, and make three test dummy files for each OS, to run and build on each VM, this way you can ensure that code will run on all major OSes.

> as for example the virtual cam only needs to be added once per boot.

Basically this can be bundled into its own file, and in the main python code if OS==OSX then run some shell script that fetches and installs virtualcamera and create a sim link in some dir easier to work with, probably inside `/virtual_webcam_background` would make more sense. One thing I don't remember is when installing virtualcamera if it requires user pass, if not then it's even better otherwise, would need to take that into account.

> if I can add your article as markdown to the project,

Yeah, go ahead, I think the whole blog post is written in markdown as well.
